src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void createNewArray() {
+        int newSize = array.length + array.length / 2;
+        T[] newArray = (T[]) new Object[newSize];
+        System.arraycopy(array, 0, newArray, 0, newSize);
+        array = newArray;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            createNewArray();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
 
+        if (index < 0 && index > size) {
+            throw new ArrayListIndexOutOfBoundsException("This index no existing in this Array");
+        }
+        if (size == array.length) {
+            createNewArray();
+        }
+        array[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int newSize = size + list.size();
+        T[] newArray = (T[]) new Object[newSize];
+        System.arraycopy(array, 0, newArray, 0, newSize);
+        array = newArray;
```

Comment:
The logic in `addAll` method is incorrect. You are not adding the elements from the provided list to the `array`. You should first check if the array needs to grow, then use `System.arraycopy` to copy elements from the list into the `array` starting at index `size`. The loop condition in line 46 is also incorrect; it should be `i < list.size()`. This is a bug in the `addAll` method.